packages/compiler/core/type-utils.ts

```diff
@@ -75,3 +79,50 @@ export function isGlobalNamespace(
 ): namespace is Namespace & { name: ""; namespace: undefined } {
   return program.getGlobalNamespaceType() === namespace;
 }
+
+/**
+ * Check if the given type is contained in the specified namespace or, optionally, its child namespaces.
+ * @param type Type
+ * @param namespace Namespace
+ * @returns {boolean}
+ */
+export function isDefinedInNamespace(
+  type: Model | Operation | Interface | Namespace | Enum,
+  namespace: Namespace,
+  options: { recursive?: boolean } = { recursive: true }
+) {
+  let collection: any[] = [];
+  switch (type.kind) {
+    case "Model":
+      collection = Array.from(namespace.models.values());
+      break;
+    case "Operation":
+      collection = Array.from(namespace.operations.values());
+      break;
+    case "Namespace":
+      collection = Array.from(namespace.namespaces.values());
+      break;
+    case "Interface":
+      collection = Array.from(namespace.interfaces.values());
+      break;
+    case "Enum":
+      collection = Array.from(namespace.enums.values());
```

Comment:
Missed this earlier, but it feels unnecessarily expensive to copy into arrays to check this. Can we not store an iterator replace that indexOf below with a simple loop.

Reply:
And maybe it would be even better to walk up through parent namespaces from type until root and check if we hit the namespace, only walking more than one level if recursive = true.

Reply:
I just realized this entire approach was really dumb, just pushed a better version of the function.

Reply:
Used `navigateTypesInNamespace`.